hoomd/snapshot.py

```diff
@@ -109,3 +109,38 @@ def replicate(self, nx, ny, nz):
 
     def _broadcast_box(self):
         self._cpp_obj._broadcast_box(self._comm.cpp_mpi_conf)
+
+    @classmethod
+    def _from_gsd_snapshot(cls, gsd_snapshot):
+        snap = cls()
+        # Set all particle attributes in snap from gsd_snapshot
+        for key in vars(gsd_snapshot.particles):
+            val = vars(gsd_snapshot.particles)[key]
+            if val is not None:
+                try:
+                    setattr(snap.particles, key, val)
+                except AttributeError:
+                    try:
+                        # Some attributes exist but aren't setable without [:]
+                        x = getattr(snap.particles,key)
+                        x[:] = val
+                    except AttributeError:
+                        # The attribute doesn't exist in hoomd.Snapshot
```

Comment:
Why is this needed? What attributes are in `gsd.hoomd.Snapshot` that are not in `hoomd.Snapshot`?

Reply:
I think the only only is `type_shapes`